stubs/setuptools/@tests/test_cases/check_distutils.py

```diff
@@ -0,0 +1,5 @@
+import distutils.command.sdist
+import distutils.config
+
```

Comment:
Maybe we should also try to use something exclusive in setuptool's version?

For example:
```suggestion
from distutils.util import split_version

s = split_version("")
```

Reply:
Whelp, it showed that checkers (at least within the context of test_cases) still prefer the stdlib if it's found. (this now only passes on 3.12).

Since those weren't in stdlib anyway, I can live with not supporting them rather than adding more hacky workarounds to the test suite.

Reply:
That fails mypy/pyright on Python <3.12, because both of them choose the stdlib `distutils` stubs over the third-party `distutils` stubs. That's the correct behaviour for any normal distribution; `setuptools` only does its "override the stdlib stubs" magic through some importlib hacks, and unfortunately I think it's unreasonable to expect type checkers to add support for that

Reply:
You could write this as `if sys.version_info >= (3, 12): split_version("")`. No opinion on whether it's needed though.

Reply:
> You could write this as `if sys.version_info >= (3, 12): split_version("")`.

Eh, I feel like that could be more confusing than clarifying, since it _is_ actually available on <py312 if you have `setuptools` installed, there's just no way of reasonably getting type checkers to understand that